internal/controlplane/handlers_githubwebhooks.go

```diff
@@ -237,82 +444,496 @@ func (s *Server) HandleGitHubWebHook() http.HandlerFunc {
+	if err != nil {
+		log.Printf("error instantiating provider: %v", err)
+		return nil, err
+	}
+
+	cli, err := provifv1.As[provifv1.GitHub](provider)
+	if err != nil {
+		log.Printf("error instantiating provider: %v", err)
+		return nil, err
+	}
+
+	tempArtifact, err := gatherArtifact(ctx, cli, event)
+	if err != nil {
+		return nil, fmt.Errorf("error gathering versioned artifact: %w", err)
+	}
+
+	dbArtifact, err := s.store.UpsertArtifact(ctx, db.UpsertArtifactParams{
+		RepositoryID: uuid.NullUUID{
+			UUID:  dbrepo.ID,
+			Valid: true,
+		},
+		ArtifactName:       tempArtifact.GetName(),
+		ArtifactType:       tempArtifact.GetTypeLower(),
+		ArtifactVisibility: tempArtifact.Visibility,
+		ProjectID:          dbrepo.ProjectID,
+		ProviderID:         dbrepo.ProviderID,
+		ProviderName:       dbrepo.Provider,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("error upserting artifact: %w", err)
+	}
+
+	_, pbArtifact, err := artifacts.GetArtifact(ctx, s.store, dbrepo.ProjectID, dbArtifact.ID)
+	if err != nil {
+		return nil, fmt.Errorf("error getting artifact with versions: %w", err)
+	}
+	// TODO: wrap in a function
+	pbArtifact.Versions = tempArtifact.Versions
+
+	eiw := entities.NewEntityInfoWrapper().
+		WithArtifact(pbArtifact).
+		WithProviderID(dbrepo.ProviderID).
+		WithProjectID(dbrepo.ProjectID).
+		WithRepositoryID(dbrepo.ID).
+		WithArtifactID(dbArtifact.ID).
+		WithActionEvent(*event.Action)
+
+	return &processingResult{topic: events.TopicQueueEntityEvaluate, eiw: eiw}, nil
+}
+
+func (s *Server) processRelevantRepositoryEvent(
+	ctx context.Context,
+	payload []byte,
+) (*processingResult, error) {
+	var event *repoEvent
+	if err := json.Unmarshal(payload, &event); err != nil {
+		return nil, err
+	}
+
+	// Check fields mandatory for processing the event
+	if event.GetRepo() == nil {
+		return nil, errRepoNotFound
+	}
+
+	l := zerolog.Ctx(ctx).With().
+		Str("github-event-action", event.GetAction()).
+		Int64("github-repository-id", event.GetRepo().GetID()).
+		Str("github-repository-url", event.GetRepo().GetHTMLURL()).
+		Logger()
+
+	if event.GetRepo().GetID() == 0 {
+		return nil, errors.New("invalid repo: id is 0")
+	}
+
+	l.Info().Msg("handling event for repository")
+
+	dbrepo, err := s.fetchRepo(ctx, event.GetRepo())
+	if err != nil {
+		return nil, err
+	}
+
+	// This only makes sense for "meta" event type
+	if dbrepo.WebhookID.Valid {
+		// Check if the payload webhook ID matches the one we
+		// have stored in the DB for this repository
+		if event.GetHookID() != dbrepo.WebhookID.Int64 {
+			// This means we got a deleted event for a
+			// webhook ID that doesn't correspond to the
+			// one we have stored in the DB.
+			return nil, newErrNotHandled("meta event with action %s not handled, hook ID %d does not match stored webhook ID %d",
+				event.GetAction(),
+				event.GetHookID(),
+				dbrepo.WebhookID.Int64,
+			)
+		}
+	}
+	// If we get this far it means we got a deleted event for a
```

Comment:
this comment seems like it's no longer relevant (or should be moved to where we handle `webhookActionEventDeleted`

Reply:
Removing it.